src/TRestRawBaseLineCorrectionProcess.cxx

```diff
@@ -0,0 +1,114 @@
+ /// 
+ /// \htmlonly <style>div.image img[src="doc_TRestRawBaseLineCorrectionProcess2.png"]{width:750px;}</style> \endhtmlonly
+ ///
+ /// ![A raw signal with and without TRestBaseLineCorrectionProcess with smoothing window size 75](doc_TRestRawBaseLineCorrectionProcess2.png) 
+ /// <hr>
+ ///
+ /// \warning **⚠ REST is under continous development.** This documentation
+ /// is offered to you by the REST community. Your HELP is needed to keep this code
+ /// up to date. Your feedback will be worth to support this software, please report
+ /// any problems/suggestions you may find while using it at [The REST Framework
+ /// forum](http://ezpc10.unizar.es). You are welcome to contribute fixing typos,
+ /// updating
+ /// information or adding/proposing new contributions. See also our
+ /// <a href="https://github.com/rest-for-physics/framework/blob/master/CONTRIBUTING.md">Contribution
+ /// Guide</a>.
+ ///
+ ///
+ ///--------------------------------------------------------------------------
+ ///
+ /// RESTsoft - Software for Rare Event Searches with TPCs
+ ///
+ /// History of developments:
+ ///
+ /// 2022-Mar:  First implementation
+ ///             Konrad Altenmueller                                                                
+ /// \class TRestRawBaseLineCorrectionProcess                                             
+ /// \author     Konrad Altenmueller
+ ///                                                                      
+ /// <hr>                                                                 
+ /////////////////////////////////////////////////////////////////////////
+
+#include "TRestRawBaseLineCorrectionProcess.h"
+
+ClassImp(TRestRawBaseLineCorrectionProcess);
+
+TRestRawBaseLineCorrectionProcess::TRestRawBaseLineCorrectionProcess() {
+    Initialize();
+}
+
+TRestRawBaseLineCorrectionProcess::~TRestRawBaseLineCorrectionProcess() {
+    delete fInputEvent;
+    delete fOutputEvent;
+}
+
+void TRestRawBaseLineCorrectionProcess::Initialize() {
+    SetSectionName(this->ClassName());
+    fInputEvent = NULL;
+    fOutputEvent = new TRestRawSignalEvent();
+}
+
+void TRestRawBaseLineCorrectionProcess::InitProcess() {
+    if (fSignalsRange.X() != -1 && fSignalsRange.Y() != -1) fRangeEnabled = true;
+}
+
+TRestEvent* TRestRawBaseLineCorrectionProcess::ProcessEvent(TRestEvent * evInput) {
+
+        fInputEvent = (TRestRawSignalEvent*)evInput;
+
+        for (int s = 0; s < fInputEvent->GetNumberOfSignals(); s++) {
+            TRestRawSignal* sgnl = fInputEvent->GetSignal(s);
+
+            if (fRangeEnabled && (sgnl->GetID() < fSignalsRange.X() || sgnl->GetID() > fSignalsRange.Y())){
+                fOutputEvent->AddSignal(*sgnl);
+                continue;
+            }
+
+            TRestRawSignal* sgnl2 = new TRestRawSignal();
+            int sID = sgnl->GetID();
+            
+            sgnl->GetBaseLineCorrected(sgnl2,fSmoothingWindow);
+            sgnl2->SetID(sID);
+            fOutputEvent->AddSignal(*sgnl2);
```

Comment:
I think we should avoid the use of `new`

```suggestion
            TRestRawSignal sgnl2( ) ;
            sgnl->GetBaseLineCorrected(&sgnl2, fSmoothingWindow);
            sgnl2.SetID( sngl->GetID( ) );
            fOutputEvent->AddSignal( sgnl2 );
```

Reply:
It doesn't compile using your suggest code here...

Reply:
Sorry, there was a bug, indeed it is difficult to compile a code suggestion, just commited some changes to avoid `new`